rcl_action/src/rcl_action/action_server.c

```diff
@@ -916,14 +916,43 @@ rcl_action_server_is_valid(const rcl_action_server_t * action_server)
   return true;
 }
 
+bool
+rcl_action_server_is_valid_except_context(const rcl_action_server_t * action_server)
+{
+  RCL_CHECK_FOR_NULL_WITH_MSG(action_server, "action server pointer is invalid", return false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    action_server->impl, "action server implementation is invalid", return false);
+  if (!rcl_service_is_valid(&action_server->impl->goal_service)) {
+    RCL_SET_ERROR_MSG("goal service is invalid");
+    return false;
+  }
+  if (!rcl_service_is_valid(&action_server->impl->cancel_service)) {
+    RCL_SET_ERROR_MSG("cancel service is invalid");
+    return false;
+  }
+  if (!rcl_service_is_valid(&action_server->impl->result_service)) {
+    RCL_SET_ERROR_MSG("result service is invalid");
+    return false;
+  }
+  if (!rcl_publisher_is_valid_except_context(&action_server->impl->feedback_publisher)) {
+    RCL_SET_ERROR_MSG("feedback publisher is invalid");
+    return false;
+  }
+  if (!rcl_publisher_is_valid_except_context(&action_server->impl->status_publisher)) {
+    RCL_SET_ERROR_MSG("status publisher is invalid");
+    return false;
+  }
+  return true;
```

Comment:
Could deduplicate with `rcl_action_server_is_valid()`.

Reply:
I'll do this on a follow up PR